docs/hive-integration/kube-api-webhook.md

```diff
@@ -0,0 +1,49 @@
+# Webhook validation
+
+Admission webhooks are HTTP callbacks that receive admission requests and do something with them. There are two types of admission webhooks: validating admission webhook and mutating admission webhook.
+
+- Mutating Admission Webhook: These can mutate the object while it’s being created or updated, before it gets stored. It can be used to default fields in a resource requests, e.g. fields in Deployment that are not specified by the user.
+
+- Validating Admission Webhook: These can validate the object while it’s being created or updated, before it gets stored. It allows more complex validation than pure schema-based validation. e.g. cross-field validation.
+
+Read more about admission webhooks [here](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
+
+Webhook validation enable adding validations based on the state of the CRDs.
+For example, changing the Spec of AgentClusterInstall or Agent after the installation started is not permitted.
+
+## Implementation
```

Comment:
The rest of this doc looks relevant only to developer environments. We'll never expect an end-user to deploy a webhook like this, or even have much awareness that webhooks exist. How about moving this doc into https://github.com/openshift/assisted-service/tree/master/docs/dev ?

Reply:
Done